google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerImpl.java

```diff
@@ -1238,12 +1244,18 @@ void backoffSleep(Context context, long backoffMillis) {
     @Nullable
     @Override
     public <T> T run(TransactionCallable<T> callable) {
+      if (inTransaction.get() == Boolean.TRUE) {
```

Comment:
can we avoid the call to begin a txn also ? the check here will only disallow it to be executed.

Reply:
Do you mean have the check in more than one place? It seems to me that we should do it in one place. If beginTransaction() is a better place, then we should move it there and get rid of the check here.
beginTransaction() seems to be at the Session level. Is a session allowed to have only one pending transaction?